lib/foreman/renderer.rb

```diff
@@ -48,12 +51,23 @@ def host_param_false?(name)
 
     def render_safe(template, allowed_methods = [], allowed_vars = {})
       if Setting[:safemode_render]
-        box = Safemode::Box.new self, allowed_methods
-        box.eval(ERB.new(template, nil, '-').src, allowed_vars)
+        # erb.src trims comments and adds extra lines :-/ no way to get the right line number at least we set the template name
+        box = Safemode::Box.new self, allowed_methods, template_name
+        erb = ERB.new(template, nil, '-')
+        box.eval(erb.src, allowed_vars)
       else
         allowed_vars.each { |k,v| instance_variable_set "@#{k}", v }
-        ERB.new(template, nil, '-').result(binding)
+        erb = ERB.new(template, nil, '-')
+        # erb allows to set location since Ruby 2.2
+        erb.location = template_name, 0 if erb.respond_to?(:location=)
+        erb.result(binding)
       end
+
+    rescue ::Racc::ParseError, ::SyntaxError => e
+      # Racc::ParseError is raised in safe mode, SyntaxError in unsafe mode
+      new_e = Foreman::Renderer::SyntaxError.new(_("Syntax error occurred on parsing the template, make sure you have all ERB tags properly closed and the Ruby syntax is valid. The Ruby error: ") + e.message)
```

Comment:
1. Untranslated strings (N_()) must be given as the first parameter to a Foreman::Exception, interpolated values passed in as additional arguments.
1. "occurred _while_ parsing the template"